a3fe/run/simulation.py

```diff
@@ -278,45 +263,24 @@ def _select_input_files(self) -> None:
     def _update_simfile(self) -> None:
         """Set the lambda value in the simulation file, as well as some
         paths to input files."""
-
-        # Check that the lambda value has been set
-        if not hasattr(self, "lam"):
-            raise AttributeError("Lambda value not set for simulation")
-
-        # Check that the set lambda value is in the list of lamvals in the simfile
-        lamvals = [
-            float(lam_val)
-            for lam_val in _read_simfile_option(
-                self.simfile_path, "lambda array"
-            ).split(",")
-        ]
-        if self.lam not in lamvals:
-            raise ValueError(
-                f"Lambda value {self.lam} not in list of lambda values in simfile"
-            )
-
-        # Set the lambda value in the simfile
-        _write_simfile_option(self.simfile_path, "lambda_val", str(self.lam))
-
-        # Set the paths to the input files
-        input_paths = {
-            "morphfile": "somd.pert",
-            "topfile": "somd.prm7",
-            "crdfile": "somd.rst7",
-        }
-
-        for option, name in input_paths.items():
-            _write_simfile_option(
-                self.simfile_path, option, _os.path.join(self.input_dir, name)
-            )
-
-        # Add the restraints file if it exists
-        if _os.path.isfile(_os.path.join(self.input_dir, "restraint.txt")):
-            with open(_os.path.join(self.input_dir, "restraint.txt"), "r") as f:
-                restraint = f.readlines()[0].split("=")[1]
-            _write_simfile_option(
-                self.simfile_path, "boresch restraints dictionary", restraint
-            )
+        
+        # update engine_config
+        self.engine_config.lambda_val = self.lam
+        self.engine_config.morphfile = _os.path.join(self.input_dir, "somd.pert")
+        self.engine_config.topfile = _os.path.join(self.input_dir, "somd.prm7")
+        self.engine_config.crdfile = _os.path.join(self.input_dir, "somd.rst7")
+        
+        # if restraint file exists, read and set
+        restraint_file = _os.path.join(self.input_dir, "restraint.txt")
+        if _os.path.isfile(restraint_file):
+            with open(restraint_file, "r") as f:
+                content = f.read().strip()
+                if "=" in content:
+                    restraint = content.split("=", 1)[1].strip()
+                    self.engine_config.boresch_restraints_dictionary = restraint
+
+        # Write updated config to file
+        self.engine_config.write_somd_config(self.output_dir, self.lam)
```

Comment:
Please rename to write_config to make it general, rather than write_somd_config - the base engine config should have an write_config abstract method, which is implemented by the somd and gromacs config objects, and then hopefully we can call them interchangably.

Reply:
Also, rather than storing the morphfile, topfile, crdfile, and lam value/ array in the engine config, let's just pass them to the write_config function.